dffml/util/net.py

```diff
@@ -145,106 +144,20 @@ def sync_urlretrieve_and_validate(
     ):
         if not target_path.parent.is_dir():
             target_path.parent.mkdir(parents=True)
-        sync_urlretrieve(
+        path, _ = sync_urlretrieve(
             url,
             filename=str(target_path),
             protocol_allowlist=protocol_allowlist,
             reporthook=progress_reporthook,
         )
+        return path
     validate_file_hash(
         target_path, expected_sha384_hash=expected_sha384_hash,
     )
+    return str(target_path.absolute())
```

Comment:
It looks like we previously had `cached_download()` giving a `pathlib.Path` object, let's maintain that

```suggestion
    return target_path.absolute()
```